http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/AutoDeleteAppMonitor.scala

```diff
@@ -0,0 +1,134 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import cats.Show
+import cats.effect.Async
+import cats.effect.std.Queue
+import cats.mtl.Ask
+import cats.syntax.all._
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.leonardo.config.AutoDeleteConfig
+import org.broadinstitute.dsde.workbench.leonardo.db._
+import org.broadinstitute.dsde.workbench.leonardo.http.dbioToIO
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoAuthProvider
+import org.broadinstitute.dsde.workbench.leonardo.model.SamResource.AppSamResource
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.DeleteAppMessage
+import org.broadinstitute.dsde.workbench.leonardo.{
+  AllowedChartName,
+  AppId,
+  AppName,
+  AppStatus,
+  CloudContext,
+  SamResourceId
+}
+import org.broadinstitute.dsde.workbench.model.{TraceId, WorkbenchEmail}
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.broadinstitute.dsp.ChartName
+import org.typelevel.log4cats.StructuredLogger
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+
+/**
+ * This monitor periodically sweeps the Leo database and auto delete App that have been running for too long.
+ */
+class AutoDeleteAppMonitor[F[_]](
+  config: AutoDeleteConfig,
+  publisherQueue: Queue[F, LeoPubsubMessage],
+  authProvider: LeoAuthProvider[F]
+)(implicit
+  dbRef: DbReference[F],
+  logger: StructuredLogger[F],
+  ec: ExecutionContext,
+  openTelemetry: OpenTelemetryMetrics[F]
+) extends BackgroundProcess[F, AppToAutoDelete] {
+  override def name: String =
+    "autoDeleteApp" //
+  override def interval: scala.concurrent.duration.FiniteDuration = config.autoDeleteCheckInterval
+
+  override def getCandidates(now: Instant)(implicit
+    F: Async[F],
+    metrics: OpenTelemetryMetrics[F],
+    logger: StructuredLogger[F]
+  ): F[Seq[AppToAutoDelete]] = for {
+    candidates <- appQuery.getAppsReadyToAutoDelete.transaction
+  } yield candidates
+
+  override def action(a: AppToAutoDelete, traceId: TraceId, now: Instant)(implicit F: Async[F]): F[Unit] =
+    for {
+      now <- F.realTimeInstant
+      loggingCtx = Map("traceId" -> traceId.asString)
+      _ <- a.cloudContext match {
+        case CloudContext.Gcp(googleProject) =>
+          if (a.appStatus == AppStatus.Error) {
```

Comment:
Does app status Error imply the corresponding Google cloud resources do not exist?

Reply:
I am not 100% sure how Leo deleting app works
I copied this delete app logic from DeleteApp endpoint logic: https://github.com/DataBiosphere/leonardo/blob/bd48425dab3c6cd40f6ed21d1c997f0db35f39cd/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala#L401L425

Reply:
I was wondeirng the same thing..but for now, I think keeping the same logic is safe to do